extensions/json-jackson/src/main/java/io/deephaven/json/jackson/ObjectMixin.java

```diff
@@ -0,0 +1,421 @@
+            this.map = allCaseSensitive()
+                    ? new HashMap<>()
+                    : new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+            for (Entry<ObjectFieldOptions, ValueProcessor> e : fields.entrySet()) {
+                final ObjectFieldOptions field = e.getKey();
+                map.put(field.name(), field);
+                for (String alias : field.aliases()) {
+                    map.put(alias, field);
+                }
+            }
+        }
+
+        private ObjectFieldOptions lookupField(String fieldName) {
+            final ObjectFieldOptions field = map.get(fieldName);
+            if (field == null) {
+                return null;
+            }
+            if (!field.caseSensitive()) {
+                return field;
+            }
+            // Need to handle the case where some fields are case-insensitive, but this one is _not_.
+            if (field.name().equals(fieldName) || field.aliases().contains(fieldName)) {
+                return field;
+            }
+            return null;
+        }
+
+        private ValueProcessor processor(ObjectFieldOptions options) {
+            return Objects.requireNonNull(fields.get(options));
+        }
+
+        @Override
+        public void processCurrentValue(JsonParser parser) throws IOException {
+            // see com.fasterxml.jackson.databind.JsonDeserializer.deserialize(com.fasterxml.jackson.core.JsonParser,
+            // com.fasterxml.jackson.databind.DeserializationContext)
+            // for notes on FIELD_NAME
+            switch (parser.currentToken()) {
+                case START_OBJECT:
+                    if (parser.nextToken() == JsonToken.END_OBJECT) {
+                        processEmptyObject(parser);
+                        return;
+                    }
+                    if (!parser.hasToken(JsonToken.FIELD_NAME)) {
+                        throw new IllegalStateException();
+                    }
+                    // fall-through
+                case FIELD_NAME:
+                    processObjectFields(parser);
+                    return;
+                case VALUE_NULL:
+                    processNullObject(parser);
+                    return;
+                default:
+                    throw Parsing.mismatch(parser, Object.class);
+            }
+        }
+
+        @Override
+        public void processMissing(JsonParser parser) throws IOException {
+            if (!allowMissing()) {
+                throw Parsing.mismatchMissing(parser, Object.class);
+            }
+            for (ValueProcessor value : fields.values()) {
+                value.processMissing(parser);
+            }
+        }
+
+        private void processNullObject(JsonParser parser) throws IOException {
+            if (!allowNull()) {
+                throw Parsing.mismatch(parser, Object.class);
+            }
+            for (ValueProcessor value : fields.values()) {
+                value.processCurrentValue(parser);
```

Comment:
Would it be better (less branchy) to implement a "processNull" API?

Reply:
It doesn't fit in as well with the data model; the reason that missing is separated b/c it's a special case, whereas null is a "normal" case (a well-typed JSON value of `null`).